### ISSUE TITLE
Custom rule new line issue

### ISSUE THREAD

**Issue by zntfdr** [realm/SwiftLint#2988]
I'm adding a few custom rules and it seems I can't find the correct way to determine when there's **not** a new empty line after a match.

I've tried multiple regexes, the one specified below can be tested [here](https://regex101.com/r/AYLXTP/1).

swiftlint conf:
```
custom_rules:
  mark_spacing_after:
    included: ".*\\.swift"
    name: "Consistent MARK spacing"
    regex: "( +// MARK: .*\n^[^\n])"
    message: "Leave an empty line after a MARK declation."
    severity: warning
```
file example:
```swift
class Example {

    // MARK: ExpressibleByDictionary

    typealias Key = String

    // MARK: ExpressibleByDictionary
    typealias KeySecond = String

}

```

Expected result:
```swift
class Example {

    // MARK: ExpressibleByDictionary

    typealias Key = String

    // MARK: ExpressibleByDictionary
^ breaking mark_spacing_after rule
    typealias KeySecond = String

}

```

Actual result:
```swift
class Example {

    // MARK: ExpressibleByDictionary
^ breaking mark_spacing_after rule

    typealias Key = String

    // MARK: ExpressibleByDictionary
    typealias KeySecond = String

}

```

Any input on how to solve this is more than welcome! 🤗 

PS
running with swiftlint 0.38

**Comment by marcelofabri** [comment]
The problem is that SwiftLint runs the regexes with the "singleline" (from your tool) frag - i.e. "dots matches newlines"

**Comment by zntfdr** [comment]
Thank you @marcelofabri, I can reproduce as you said by enabling the singleline flag (new test [here](https://regex101.com/r/AYLXTP/2)):
will investigate and report this evening!

**Comment by zntfdr** [comment]
Thank you again @marcelofabri, your help led me to the fix, this is the working regex: `( +\/\/ MARK: [^\n]*\n^[^\n])`.

And here's the [updated (working!) test/demo](https://regex101.com/r/cjOEtc/3) 🎉 

Wish you a wonderful day!

**Comment by marcelofabri** [comment]
Glad you could make it work, thanks for the update!
